### ISSUE TITLE
Three.js cannot work with Tern.

### ISSUE THREAD

**Issue by yiakwy** [mrdoob/three.js#13020]
##### Description of the problem
Most npm modules can work with Tern (tern_for_vim), but I found tern_for_vim cannot complete three.js when I import three using ES6

> import * as TRHEE from "three"
##### Three.js version

- [ ] Dev
- [x] r89
- [ ] ...

##### Browser

- [x] All of them
- [x] Chrome
- [ ] Firefox
- [ ] Internet Explorer

##### OS

- [x] All of them
- [ ] Windows
- [x] macOS
- [ ] Linux
- [ ] Android
- [ ] iOS

**Comment by looeee** [comment]
It seems likely that this is an issue you should raise with the tern_for_vim developers.

**Comment by Mugen87** [comment]
I'm not familiar with tern but there is a tern file which is used in the editor (see [here](https://github.com/mrdoob/three.js/blob/dev/editor/js/libs/tern-threejs/threejs.js)).

But be careful. This file is not really maintained and does not reflect the entire API.

**Comment by yiakwy** [comment]
Hi @looeee I also reported the problem to tern developers.  I use vim in server side and try to build a modern ES development environment within vim. It was true (C, python, es6 with other modules) until I use three.js. It takes me a lot of time before I figure out that there might be some conflicts between three.js and tern_for_vim.

@Mugen87 Thanks for your help.  [tern_for_vim](https://github.com/ternjs/tern_for_vim) is vim plugin so that it can complete semantically what I typed in vim.
